chcemvediet/apps/obligees/tests/test_forms.py

```diff
@@ -1,123 +1,175 @@
+        self.assertEqual(city.text, u'ccc city')
+        email = details.find_class(u'chv-obligee-widget-email')[0]
+        self.assertEqual(email.text, u'ccc@a.com')
 
-    @unittest.skip(u'FIXME')
     def test_new_form_with_initial_value_as_obligee_name(self):
         names = [u'aaa', u'bbb', u'ccc', u'ddd']
-        oblgs = [self._create_obligee(name=n) for n in names]
+        oblgs = [self._create_obligee(name=n, street=u'{} street'.format(n), city=u'{} city'.format(n), zip=u'12345', emails=u'{}@a.com'.format(n)) for n in names]
         form = self.Form(initial={u'obligee': u'ccc'})
         rendered = self._render(u'{{ form }}', form=form)
+        html = lxml.html.fromstring(rendered)
         self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text" value="ccc">
+                <input class="form-control pln-autocomplete" data-autocomplete-url="{url}" name="obligee" data-name="obligee" type="text" value="ccc">
                 """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        details = html.find_class(u'chv-obligee-widget-details')[0]
+        street = details.find_class(u'chv-obligee-widget-street')[0]
+        self.assertEqual(street.text, u'ccc street')
+        zip = details.find_class(u'chv-obligee-widget-zip')[0]
+        self.assertEqual(zip.text, u'12345')
+        city = details.find_class(u'chv-obligee-widget-city')[0]
+        self.assertEqual(city.text, u'ccc city')
+        email = details.find_class(u'chv-obligee-widget-email')[0]
+        self.assertEqual(email.text, u'ccc@a.com')
 
-    @unittest.skip(u'FIXME')
     def test_submitted_with_empty_value_but_required(self):
         form = self.Form({u'obligee': u''})
         self.assertFalse(form.is_valid())
         self.assertEqual(form.errors[u'obligee'], [u'This field is required.'])
 
         rendered = self._render(u'{{ form }}', form=form)
         self.assertInHTML(u'<ul class="errorlist"><li>This field is required.</li></ul>', rendered)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
 
-    @unittest.skip(u'FIXME')
     def test_submitted_with_empty_value_but_not_required(self):
         form = self.Form({u'obligee': u''})
         form.fields[u'obligee'].required = False
         self.assertTrue(form.is_valid())
         self.assertIsNone(form.cleaned_data[u'obligee'])
 
         rendered = self._render(u'{{ form }}', form=form)
-        self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
-                """.format(url=reverse(u'obligees:autocomplete')), rendered)
+        self.assertInHTML(u'<ul class="errorlist"><li>This field is required.</li></ul>', rendered, count=0)
```

Comment:
Nie je nieco ako `self.assertNotInHTML`? Lebo `assertInHTML` s `count=0` je divne.

Reply:
Mozeme si napisat vlastnu funkciu `assertNotInHtml`, ktora funkcnostou bude `assertInHtml` s paramentrom `count=0`. Alebo riesenie s pomocou `lxml` + `assertIsNone(element)`.

Reply:
Jj, takto cez `lxml` je to prehladnejsie. Asi do buducna budeme vacsinu HTML assertov robit takto. Okrem nejakych trivialnych mozno.